src/System.Net.HttpListener/src/System/Net/Windows/WebSockets/HttpWebSocket.Windows.cs

```diff
@@ -320,35 +221,13 @@ internal static void ThrowPlatformNotSupportedException_WSPC()
             throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
         }
 
-        private static void ThrowPlatformNotSupportedException_HTTPSYS()
-        {
-            throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
-        }
-
-        internal static void ValidateArraySegment<T>(ArraySegment<T> arraySegment, string parameterName)
-        {
-            Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
-
-            if (arraySegment.Array == null)
-            {
-                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
-            }
-
-            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
-            {
-                throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Offset));
-            }
-            if (arraySegment.Count < 0 || arraySegment.Count > (arraySegment.Array.Length - arraySegment.Offset))
-            {
-                throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Count));
-            }
-        }
+        private static string SupportedVersion => WebSocketProtocolComponent.SupportedVersion;
 
-        private static void EnsureHttpSysSupportsWebSockets()
+        private static void ValidateWebSocketHeadersCore(HttpListenerContext context)
```

Comment:
Or alternatively have a WebSocketsSupported property, and have the PNSE throw be at the call site: then the managed implementation would always return true and the Windows would return s_httpSysSupportsWebSockets.